### PR TITLE
Static analyzers: Address some parameter-can-be-const analytics

### PR DESCRIPTION
**Pull request by akva2**
A small drip of addressing static analyzer analytics.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/output/eclipse/AggregateActionxData.cpp [hunk 1]
```diff
@@ -378,6 +377,11 @@ namespace {
                 return this->sacn_.get()(this->action_, condition);
             }
 
+            decltype(auto) operator[](const Matrix::Idx condition) const
+            {
+                return this->sacn_.get()(this->action_, condition);
+            }
```
Comment (bska) [inline opm/output/eclipse/AggregateActionxData.cpp:383]: It doesn't hurt, obviously, but it's kind of misleading anyway.  This "just" means the `reference_wrapper` itself is const, but the underlying `Matrix` contained in that `reference_wrapper` is still mutable.

File: opm/output/eclipse/AggregateActionxData.cpp [hunk 2]
```diff
@@ -424,7 +428,7 @@ namespace {
                            const Opm::SummaryState&    st,
                            const Opm::Schedule&        sched,
                            const std::size_t           simStep,
-                           Array&                      sAcn)
+                           const Array&                sAcn)
```
Comment (bska) [inline opm/output/eclipse/AggregateActionxData.cpp:431]: I appreciate it, but I'd prefer that this remain non-`const` here.  The code structure **probably** defeats static analysis, but the `sAcn` object really is being mutated here.  In particular, the return value from `Array::operator[]()` is a view into a mutable range, so when we create the per-condition [`sacn`](https://github.com/OPM/opm-common/blob/cd7e2a2c65138a41db19f509bcba4c73c5389ad7/opm/output/eclipse/AggregateActionxData.cpp#L442) object and assign values to it, we're really changing the underlying `Array` object.

Not marking `sAcn` as `const` is intended to signify that fact, but I would really like a better way of doing it.  I'm hopeful that C++20's `std::span<>` will make these semantics more transparent...

**Review by bska — COMMENTED** [review]
This looks good for the most part, but I'm on the fence with respect to the changes in `AggregateActionxData.cpp`

File: opm/output/eclipse/AggregateActionxData.cpp (hunk 2 above)
Comment (akva2) [reply opm/output/eclipse/AggregateActionxData.cpp:431]: done. commit removed.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**